corehq/apps/app_manager/suite_xml/post_process/remote_requests.py

```diff
@@ -292,12 +286,49 @@ def get_smart_link_function(self):
         app_id = self.app.upstream_app_id if is_linked_app(self.app) else self.app.origin_id
         url = absolute_reverse("session_endpoint", args=["---", app_id, self.module.session_endpoint_id])
         prefix, suffix = url.split("---")
-        return f"concat('{prefix}', $domain, '{suffix}')"
+        params = ""
+        argument_ids = self.endpoint_argument_ids
+        if argument_ids:
+            params = f", '?{argument_ids[-1]}=', ${argument_ids[-1]}"
+            for argument_id in argument_ids[:-1]:
+                params = f", '&{argument_id}=', ${argument_id}"
```

Comment:
I don't think this works since each iteration of the loop overwrites the params var. Perhaps it should be `params +=`

```suggestion
                params += f", '&{argument_id}=', ${argument_id}"
```